### DISCUSSION TITLE
Pico W Bluetooth

### DISCUSSION THREAD

**Discussion by Guustaaf** [micropython/micropython#11121] [RP2040 / Pico]
To whoever is working on uP bluetooth for the PicoW, I would like to sincerely express my gratitude and encouragement. :)

**Comment by bidrohini** [comment]
Has there been any update after this? https://www.makeuseof.com/ihow-to-try-bluetooth-on-the-raspberry-pi-pico-w/

**Reply by mattytrentini** [reply]
Yes, Damien provided an update at our last Melbourne MicroPython Meetup:

https://www.youtube.com/watch?v=fH33D6ER7ls

In short, there has been a bug identified when using WiFi and BLE simultaneously on the Pico W. While we wait for a fix from the manufacturer, MicroPython v1.20 will go ahead with a workaround to allow only one or the other radio to be enabled. It's expected that v1.20 will be out _soon_.

**Reply by ZacharyInDallas** [reply]
I tried doing an import bluetooth on v1.20 and the latest nightly build and got ImportError: no module named 'bluetooth'.  Where is the Pico W Bluetooth workaround available?

**Reply by mattytrentini** [reply]
The workaround was found to be too problematic - v1.20 had to be shipped _without_ BLE for the Pico W. Still waiting on a fix...

**Reply by ZacharyInDallas** [reply]
Any estimates on when Pico W BLE will be available for testing?

**Reply by jimmo** [reply]
@ZacharyInDallas You already asked the same question in the PR where this work is happening. When that PR is merged then it will be ready. As you can see from the latest update in that PR, we're waiting for a [pico-sdk](https://github.com/raspberrypi/pico-sdk/) update which is up to Raspberry Pi and is out of our control. The btstack update is in https://github.com/micropython/micropython/pull/11741

Yes it will support aioble and pairing and bonding.

**Reply by ZacharyInDallas** [reply]
Thanks

**Reply by actanner** [reply]
Friends:  My first time here, but I hope my comments help any non-experts who may find it.  I was surprised to find that all versions of micropython which are referred to as 1.20.0 are not the same!   But if you get the one that shows this in the Thonny interactive window:
MicroPython v1.20.0-198-g0eacdeb1c on 2023-06-13
then the bluetooth module will load and the BLE capability will work. This is the version contained in this file:   micropython-firmware-pico-w-130623.uf2

**Reply by Josverl** [reply]
If you look at the firmware signature in more detail you will find some additional relevant information

- `v1.20.0-198-g0eacdeb1c`
-  (base) version: `v1.20.0`
- commits since that version : `198`
  So 198 differences/commits from v1.20.0
- git hash of the commit this FW was built from : `g0eacdeb1c`



The daily firmware naming however uses
micropython-firmware-pico-w-130623.uf2
- day : 13
- month : 06
- year: 23

I've not found a simple way to correlate these.

**Reply by scruss** [reply]
where are you seeing dailies without the git hash? All the ones in https://micropython.org/download/rp2-pico-w/ have the hash in the download file name
